osdslet daemon crashed when creating file share acl

<!-- This form is for bug reports and feature requests! -->

**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
When I use opensds dashboard to create a file share and then create the file share acl, the system crashed and `osdslet` service broken leaving the file share acl status to `creating`.

Here is the output log of `osdslet` service:
```shell
root@ubuntu:~/gopath/src/github.com/opensds/opensds# ./build/out/bin/osdslet -logtostderr
I0611 23:59:45.828753   14831 logs.go:75] [Info] LogFlushFrequency: 5s
I0611 23:59:45.840819   14831 controller.go:92] Controller server initialized! Start listening on port:[::]:50049
I0612 00:00:01.730634   14831 controller.go:969] controller server receive create file share acl request, vr =id:"e8948243-9a59-40ac-a31f-81bbe04471d7" fileshareId:"270748a8-4284-44c7-89e6-1b0b763957fe" type:"ip" accessCapability:"Read" accessCapability:"Write" accessTo:"10.10.3.150" profile:"{\"id\":\"09de8a13-1872-40df-be46-f1c93eeafcc2\",\"createdAt\":\"2019-06-11T22:53:52\",\"updatedAt\":\"\",\"name\":\"default\",\"description\":\"default policy\",\"storageType\":\"file\",\"provisioningProperties\":{\"dataStorage\":{\"storageAccessCapability\":[\"Read\",\"Write\",\"Execute\"],\"isSpaceEfficient\":false},\"ioConnectivity\":{}},\"replicationProperties\":{\"dataProtection\":{\"isIsolated\":false},\"replicaInfos\":{}},\"snapshotProperties\":{\"schedule\":{},\"retention\":{},\"topology\":{}},\"dataProtectionProperties\":{\"dataProtection\":{\"isIsolated\":false}}}" context:"{\"auth_token\":\"\",\"user_id\":\"558057c4256545bd8a307c37464003c9\",\"tenant_id\":\"94b280022d0c4401bcf3b0ea85870519\",\"domain_id\":\"\",\"user_domain_id\":\"\",\"project_domain_id\":\"\",\"is_admin\":true,\"read_only\":\"\",\"show_deleted\":\"\",\"request_id\":\"\",\"resource_uuid\":\"\",\"overwrite\":\"\",\"roles\":[\"admin\"],\"user_name\":\"\",\"project_name\":\"\",\"domain_name\":\"\",\"user_domain_name\":\"\",\"project_domain_name\":\"\",\"is_admin_project\":true,\"service_token\":\"\",\"service_user_id\":\"\",\"service_user_name\":\"\",\"service_user_domain_id\":\"\",\"service_user_domain_name\":\"\",\"service_project_id\":\"\",\"service_project_name\":\"\",\"service_project_domain_id\":\"\",\"service_project_domain_name\":\"\",\"service_roles\":\"\",\"token\":\"\",\"uri\":\"/v1beta/94b280022d0c4401bcf3b0ea85870519/file/acls\"}" metadata:<key:"lvPath" value:"/dev/opensds-files-default/shr_02" > metadata:<key:"nfsFileshareID" value:"270748a8-4284-44c7-89e6-1b0b763957fe" > metadata:<key:"nfsFileshareName" value:"shr_02" > 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x9c972c]

goroutine 34 [running]:
github.com/opensds/opensds/pkg/db/drivers/etcd.(*Client).UpdateFileShareAcl(0xc00017e2c0, 0xc0000fe200, 0xc0002267e0, 0x0, 0x0, 0x0)
	/root/gopath/src/github.com/opensds/opensds/pkg/db/drivers/etcd/etcd.go:281 +0x3c
github.com/opensds/opensds/pkg/controller.(*Controller).CreateFileShareAcl(0xc0001545b0, 0xd37bc0, 0xc00013bd40, 0xc00010fba0, 0x0, 0x0, 0x0)
	/root/gopath/src/github.com/opensds/opensds/pkg/controller/controller.go:1001 +0xcf4
github.com/opensds/opensds/pkg/model/proto._FileShareController_CreateFileShareAcl_Handler(0xc0f9c0, 0xc0001545b0, 0xd37bc0, 0xc00013bd40, 0xc000182730, 0x0, 0xd37bc0, 0xc00013bd40, 0xc000039200, 0x5d1)
	/root/gopath/src/github.com/opensds/opensds/pkg/model/proto/model.pb.go:5502 +0x23e
github.com/opensds/opensds/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001b2000, 0xd3ee20, 0xc0001b2180, 0xc00017a200, 0xc00013b530, 0x12d2180, 0x0, 0x0, 0x0)
	/root/gopath/src/github.com/opensds/opensds/vendor/google.golang.org/grpc/server.go:966 +0x470
github.com/opensds/opensds/vendor/google.golang.org/grpc.(*Server).handleStream(0xc0001b2000, 0xd3ee20, 0xc0001b2180, 0xc00017a200, 0x0)
	/root/gopath/src/github.com/opensds/opensds/vendor/google.golang.org/grpc/server.go:1245 +0xd25
github.com/opensds/opensds/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000037b50, 0xc0001b2000, 0xd3ee20, 0xc0001b2180, 0xc00017a200)
	/root/gopath/src/github.com/opensds/opensds/vendor/google.golang.org/grpc/server.go:685 +0x9f
created by github.com/opensds/opensds/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/root/gopath/src/github.com/opensds/opensds/vendor/google.golang.org/grpc/server.go:683 +0xa1
```

**What you expected to happen**:
Fix this bug.

**How to reproduce it (as minimally and precisely as possible)**:
Deploy opensds cluster and login in opensds dashboard, create a file share and then create a share acl, you will find `osdslet` would be broken.

**Anything else we need to know?**:

**Environment**:
- Hotpot(release/branch) version: development
- OS (e.g. from /etc/os-release): Ubuntu 16.04
- Kernel (e.g. `uname -a`):
- Install tools: `devsds`
- Others:

--- a/contrib/connector/nvmeof/nvmeof_helper.go
+++ b/contrib/connector/nvmeof/nvmeof_helper.go
@@ -30,11 +30,11 @@ const (
 
 // ConnectorInfo define
 type ConnectorInfo struct {
-	Nqn       string `mapstructure:"targetNQN"`     //NVMe subsystem name to the volume to be connected
-	TgtPort   string `mapstructure:"targetPort"`    //NVMe target port that hosts the nqn sybsystem
-	TgtPortal string `mapstructure:"targetIP"`      //NVMe target ip that hosts the nqn sybsystem
+	Nqn       string `mapstructure:"targetNQN"`    //NVMe subsystem name to the volume to be connected
+	TgtPort   string `mapstructure:"targetPort"`   //NVMe target port that hosts the nqn sybsystem
+	TgtPortal string `mapstructure:"targetIP"`     //NVMe target ip that hosts the nqn sybsystem
 	TranType  string `mapstructure:"transporType"` // Nvme transport type
-	HostNqn   string `mapstructure:"hostNqn"`       // host nqn
+	HostNqn   string `mapstructure:"hostNqn"`      // host nqn
 }
 
 //////////////////////////////////////////////////////////////////////////////////////////
@@ -146,16 +146,16 @@ func Connect(connMap map[string]interface{}) (string, error) {
 	connNqn := conn.Nqn
 	targetPortal := conn.TgtPortal
 	port := conn.TgtPort
-	nvmeTransportType := conn.TranType 
- 	hostName := conn.HostNqn
-	
-	cmd := "nvme connect -t " + nvmeTransportType + " -n " + connNqn + " -s " + port + " -a " + targetPortal
-       if hostName != "ALL" {
-               cmd += " -q " + hostName
-       }
-       //log.Printf("conn information:%s, %s, %s ", connNqn, targetPortal, port)
+	nvmeTransportType := conn.TranType
+	hostName := conn.HostNqn
 
-       _, err := connector.ExecCmd("/bin/bash", "-c", cmd)
+	cmd := "nvme connect -t " + nvmeTransportType + " -n " + connNqn + " -s " + port + " -a " + targetPortal
+	if hostName != "ALL" {
+		cmd += " -q " + hostName
+	}
+	//log.Printf("conn information:%s, %s, %s ", connNqn, targetPortal, port)
+
+	_, err := connector.ExecCmd("/bin/bash", "-c", cmd)
 
 	if err != nil {
 		log.Println("Failed to connect to NVMe nqn :", connNqn)

--- a/contrib/drivers/filesharedrivers/drivers.go
+++ b/contrib/drivers/filesharedrivers/drivers.go
@@ -38,17 +38,17 @@ type FileShareDriver interface {
 
 	CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareSpec, error)
 
-	CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error)
-
-	DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.FileShareAclSpec, error)
-
-	ListPools() ([]*model.StoragePoolSpec, error)
-
-	DeleteFileShare(opts *pb.DeleteFileShareOpts) (*model.FileShareSpec, error)
+	DeleteFileShare(opts *pb.DeleteFileShareOpts) error
 
 	CreateFileShareSnapshot(opts *pb.CreateFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error)
 
-	DeleteFileShareSnapshot(opts *pb.DeleteFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error)
+	DeleteFileShareSnapshot(opts *pb.DeleteFileShareSnapshotOpts) error
+
+	CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error)
+
+	DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error
+
+	ListPools() ([]*model.StoragePoolSpec, error)
 }
 
 // Init

--- a/contrib/drivers/filesharedrivers/manila/manila.go
+++ b/contrib/drivers/filesharedrivers/manila/manila.go
@@ -37,7 +37,7 @@ import (
 	pb "github.com/opensds/opensds/pkg/model/proto"
 	"github.com/opensds/opensds/pkg/utils/config"
 	"github.com/opensds/opensds/pkg/utils/pwd"
-	"github.com/satori/go.uuid"
+	uuid "github.com/satori/go.uuid"
 )
 
 const (
@@ -239,15 +239,15 @@ func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareS
 }
 
 // DeleteFileShare implementation
-func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) (*model.FileShareSpec, error) {
+func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) error {
 	manilaShareID := opt.Metadata[KManilaShareID]
 	if err := sharesv2.Delete(d.sharedFileSystemV2, manilaShareID).ExtractErr(); err != nil {
 		log.Error("cannot delete share:", err)
-		return nil, err
+		return err
 	}
 
 	log.V(5).Info("function DeleteFileShare succeeded\n")
-	return nil, nil
+	return nil
 }
 
 // PullFileShare implementation
@@ -318,8 +318,6 @@ func (d *Driver) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (fshare *mod
 		BaseModel: &model.BaseModel{
 			Id: opt.Id,
 		},
-
-		//TenantId:
 		FileShareId:      opt.FileshareId,
 		Type:             opt.Type,
 		AccessCapability: opt.GetAccessCapability(),
@@ -333,7 +331,7 @@ func (d *Driver) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (fshare *mod
 }
 
 // DeleteFileShareAcl implementation
-func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.FileShareAclSpec, error) {
+func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error {
 	opts := &sharesv2.RevokeAccessOpts{
 		AccessID: opt.Metadata[KManilaShareACLID],
 	}
@@ -341,11 +339,11 @@ func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.File
 	d.sharedFileSystemV2.Microversion = "2.7"
 	if err := sharesv2.RevokeAccess(d.sharedFileSystemV2, opt.Metadata[KManilaShareID], opts).ExtractErr(); err != nil {
 		log.Error("cannot revoke access:", err)
-		return nil, err
+		return err
 	}
 
 	log.V(5).Info("function DeleteFileShareAcl succeeded\n")
-	return nil, nil
+	return nil
 }
 
 // CreateFileShareSnapshot implementation
@@ -409,15 +407,15 @@ func (d *Driver) CreateFileShareSnapshot(opt *pb.CreateFileShareSnapshotOpts) (*
 }
 
 // DeleteFileShareSnapshot implementation
-func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error) {
+func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) error {
 	manilaSnapID := opt.Metadata[KManilaSnapID]
 	if err := snapshotsv2.Delete(d.sharedFileSystemV2, manilaSnapID).ExtractErr(); err != nil {
 		log.Error("cannot delete share:", err)
-		return nil, err
+		return err
 	}
 
 	log.V(5).Info("function DeleteFileShareSnapshot succeeded\n")
-	return nil, nil
+	return nil
 }
 
 // PullFileShareSnapshot implementation

--- a/contrib/drivers/filesharedrivers/nfs/nfs.go
+++ b/contrib/drivers/filesharedrivers/nfs/nfs.go
@@ -85,52 +85,68 @@ func (d *Driver) Setup() error {
 
 func (*Driver) Unset() error { return nil }
 
-func (d *Driver) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (fshare *model.FileShareAclSpec, err error) {
+func (d *Driver) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error) {
 	var access string
-	var accessCapability []string
 	// Get accessto list
 	accessTo := opt.GetAccessTo()
 	// get accessCapability list
-	accessCapability = opt.GetAccessCapability()
+	accessCapability := opt.GetAccessCapability()
 	// get fileshare name
 	fname := opt.Name
 
 	permissions := []string{"write"}
-        WriteAccess := false
+	WriteAccess := false
 
 	for _, value := range accessCapability {
 		value = strings.ToLower(value)
 		if utils.Contains(permissions, value) {
 			WriteAccess = true
 		}
-		if value == "Execute"{
+		if value == "Execute" {
 			log.Error("invalid permission:", value)
 			return nil, nil
 		}
 	}
-        if WriteAccess {
+	if WriteAccess {
 		access = AccessLevelRw
 	} else {
 		access = AccessLevelRo
 	}
 
-	err = d.cli.CreateAccess(accessTo, access, fname)
-	return fshare, nil
+	if err := d.cli.CreateAccess(accessTo, access, fname); err != nil {
+		log.Errorf("grant access %s to %s failed %v", accessTo, fname, err)
+		return nil, err
+	}
+
+	shareAccess := &model.FileShareAclSpec{
+		BaseModel: &model.BaseModel{
+			Id: opt.Id,
+		},
+		FileShareId:      opt.FileshareId,
+		Type:             opt.Type,
+		AccessCapability: accessCapability,
+		AccessTo:         accessTo,
+		Metadata:         map[string]string{},
+	}
+	return shareAccess, nil
 }
 
-func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (fshare *model.FileShareAclSpec, err error) {
-	var accessTo string
+func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error {
 	// Get accessto list
-	accessTo = opt.GetAccessTo()
+	accessTo := opt.GetAccessTo()
 	// get fileshare name
 	fname := opt.Name
 
-	err = d.cli.DeleteAccess(accessTo, fname)
+	if err := d.cli.DeleteAccess(accessTo, fname); err != nil {
+		log.Error("cannot revoke access:", err)
+		return err
+	}
 
-	return fshare, nil
+	return nil
 }
 
-func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (fshare *model.FileShareSpec, err error) {
+func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareSpec, error) {
+	var fshare *model.FileShareSpec
 	//get the server ip for configuration
 	var server = d.conf.TgtBindIp
 	//get fileshare name
@@ -147,8 +163,8 @@ func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (fshare *model.Fil
 		return nil, err
 	}
 
-	if err = d.cli.CreateVolume(name, vg, opt.GetSize()); err != nil {
-		return
+	if err := d.cli.CreateVolume(name, vg, opt.GetSize()); err != nil {
+		return nil, err
 	}
 	// remove created volume if got error
 	defer func() {
@@ -165,28 +181,26 @@ func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (fshare *model.Fil
 		log.Error("failed to create filesystem logic volume:", err)
 		return nil, err
 	}
-
 	// mount the volume to directory
 	if err := d.cli.Mount(lvPath, dirName); err != nil {
 		log.Error("failed to mount a directory:", err)
 		return nil, err
 	}
-
 	// Set permission to directory
 	if err := d.cli.SetPermission(dirName); err != nil {
 		log.Error("failed to set permission:", err)
 		return nil, err
 	}
-
 	// get export location of fileshare
 	var location []string
 	location = []string{d.cli.GetExportLocation(name, server)}
 	if len(location) == 0 {
-		log.Error("failed to get exportlocation:", err)
-		return nil, err
+		errMsg := errors.New("failed to get exportlocation: export location is empty!")
+		log.Error(errMsg)
+		return nil, errMsg
 	}
 
-	ffshare := &model.FileShareSpec{
+	fshare = &model.FileShareSpec{
 		BaseModel: &model.BaseModel{
 			Id: opt.GetId(),
 		},
@@ -202,7 +216,7 @@ func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (fshare *model.Fil
 			KLvPath:        lvPath,
 		},
 	}
-	return ffshare, nil
+	return fshare, nil
 }
 
 // ListPools
@@ -237,7 +251,7 @@ func (d *Driver) ListPools() ([]*model.StoragePoolSpec, error) {
 }
 
 // delete fileshare from device
-func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) (fshare *model.FileShareSpec, err error) {
+func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) error {
 	// get fileshare name to be deleted
 	fname := opt.GetMetadata()[KFileshareName]
 	// get fileshare path
@@ -245,22 +259,22 @@ func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) (fshare *model.Fil
 	// get directory where fileshare mounted
 	var dirName = path.Join("/var/", fname)
 
-	// unmount the volume to directory
+	// umount the volume to directory
 	if err := d.cli.UnMount(dirName); err != nil {
 		log.Error("failed to mount a directory:", err)
-		return fshare, err
+		return err
 	}
-	// dlete the actual fileshare from device
+	// delete the actual fileshare from device
 	if err := d.cli.Delete(fname, lvPath); err != nil {
 		log.Error("failed to remove logic volume:", err)
-		return fshare, err
+		return err
 	}
 	// Delete the directory
 	if err := d.cli.DeleteDirectory(dirName); err != nil {
 		log.Error("failed to delete the directory:", err)
-		return nil, err
+		return err
 	}
-	return fshare, nil
+	return nil
 }
 
 // CreateFileShareSnapshot
@@ -296,24 +310,24 @@ func (d *Driver) CreateFileShareSnapshot(opt *pb.CreateFileShareSnapshotOpts) (*
 }
 
 // DeleteFileShareSnapshot
-func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error) {
+func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) error {
 	lvsPath, ok := opt.GetMetadata()[KLvPath]
 	snapName := opt.GetMetadata()[KFileshareSnapName]
 	if !ok {
 		err := errors.New("can't find 'lvsPath' in snapshot metadata, ingnore it!")
 		log.Error(err)
-		return nil, nil
+		return nil
 	}
 	fields := strings.Split(lvsPath, "/")
 	vg := fields[2]
 	if !d.cli.Exists(snapName) {
 		log.Warningf("Snapshot(%s) does not exist, nothing to remove", snapName)
-		return nil, nil
+		return nil
 	}
 
 	if err := d.cli.DeleteFileShareSnapshots(snapName, vg); err != nil {
 		log.Error("failed to remove logic volume:", err)
-		return nil, err
+		return err
 	}
-	return nil, nil
+	return nil
 }

--- a/contrib/drivers/filesharedrivers/oceanstor/oceanstor.go
+++ b/contrib/drivers/filesharedrivers/oceanstor/oceanstor.go
@@ -28,7 +28,7 @@ import (
 	model "github.com/opensds/opensds/pkg/model"
 	pb "github.com/opensds/opensds/pkg/model/proto"
 	"github.com/opensds/opensds/pkg/utils/config"
-	"github.com/satori/go.uuid"
+	uuid "github.com/satori/go.uuid"
 )
 
 func (d *Driver) Setup() error {
@@ -343,18 +343,18 @@ func (d *Driver) GetProtoFromProfile(prf string) (string, error) {
 	return shareProto, nil
 }
 
-func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) (*model.FileShareSpec, error) {
+func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) error {
 	shareProto, err := d.GetProtoFromProfile(opt.GetProfile())
 	if err != nil {
 		log.Error(err.Error())
-		return nil, err
+		return err
 	}
 
 	meta := opt.GetMetadata()
 	if meta == nil || (meta != nil && meta[FileShareName] == "" && meta[FileShareID] == "") {
 		msg := "cannot get file share name and id"
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	fsName := meta[FileShareName]
@@ -366,21 +366,20 @@ func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) (*model.FileShareS
 	if err := shareDriver.deleteShare(shareID); err != nil {
 		msg := fmt.Sprintf("delete file share %s failed: %v", sharePath, err)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	log.Infof("delete share %s successfully", sharePath)
 
-	err = d.DeleteFileSystem(fsName)
-	if err != nil {
+	if err := d.DeleteFileSystem(fsName); err != nil {
 		msg := fmt.Sprintf("delete filesystem %s failed: %v", fsName, err)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	log.Infof("delete file system %s successfully", fsName)
 
-	return nil, nil
+	return nil
 }
 
 func (d *Driver) DeleteFileSystem(fsName string) error {
@@ -451,12 +450,12 @@ func (d *Driver) CreateFileShareSnapshot(opt *pb.CreateFileShareSnapshotOpts) (*
 	}, nil
 }
 
-func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error) {
+func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) error {
 	meta := opt.GetMetadata()
 	if meta == nil || (meta != nil && meta[FileShareSnapshotID] == "") {
 		msg := "cannot get file share snapshot id"
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	snapID := meta[FileShareSnapshotID]
@@ -465,12 +464,12 @@ func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) (*
 	if err != nil {
 		msg := fmt.Sprintf("delete filesystem snapshot %s failed, %v", snapID, err)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	log.Infof("delete file share snapshot %s successfully", snapID)
 
-	return nil, nil
+	return nil
 }
 
 func (d *Driver) getAccessLevel(accessLevels []string, shareProto string) (string, error) {
@@ -621,11 +620,11 @@ func (d *Driver) createAccessIfNotExist(shareID, accessTo, shareProto, accessLev
 	return nil
 }
 
-func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.FileShareAclSpec, error) {
+func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error {
 	shareName, shareProto, accessTo, err := d.DeleteFileShareAclParamCheck(opt)
 	if err != nil {
 		log.Error(err.Error())
-		return nil, err
+		return err
 	}
 
 	shareDriver := NewProtocol(shareProto, d.Client)
@@ -633,13 +632,13 @@ func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.File
 	share, err := shareDriver.getShare(shareName)
 	if err != nil {
 		log.Error(err.Error())
-		return nil, err
+		return err
 	}
 
 	if share == nil {
 		msg := fmt.Sprintf("share %s does not exist", shareName)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	shareID := shareDriver.getShareID(share)
@@ -648,22 +647,22 @@ func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.File
 	if err != nil {
 		msg := fmt.Sprintf("get access from share failed: %v", err)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	if accessID == "" {
 		msg := fmt.Sprintf("can not get access id from share %s", shareName)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
 	if err := d.removeAccessFromShare(accessID, shareProto); err != nil {
 		msg := fmt.Sprintf("remove access from share failed: %v", err)
 		log.Error(msg)
-		return nil, errors.New(msg)
+		return errors.New(msg)
 	}
 
-	return nil, nil
+	return nil
 }
 
 func (d *Driver) DeleteFileShareAclParamCheck(opt *pb.DeleteFileShareAclOpts) (string, string, string, error) {

--- a/contrib/drivers/huawei/dorado/metrics.go
+++ b/contrib/drivers/huawei/dorado/metrics.go
@@ -15,7 +15,6 @@
 package dorado
 
 import (
-	"github.com/satori/go.uuid"
 	"strconv"
 	"time"
 
@@ -23,6 +22,7 @@ import (
 	. "github.com/opensds/opensds/contrib/drivers/utils/config"
 	"github.com/opensds/opensds/pkg/model"
 	"github.com/opensds/opensds/pkg/utils/config"
+	uuid "github.com/satori/go.uuid"
 	"gopkg.in/yaml.v2"
 )
 

--- a/contrib/drivers/lvm/targets/nvmeof.go
+++ b/contrib/drivers/lvm/targets/nvmeof.go
@@ -188,16 +188,16 @@ func (t *NvmeoftgtTarget) CreateNvmeofTarget(volId, tgtNqn, path, initiator, tra
 		log.Infof("%s subsystem has existed", tgtNqn)
 	}
 
-	subexisted =  NvmetDir + "/subsystems/" + tgtNqn 
+	subexisted = NvmetDir + "/subsystems/" + tgtNqn
 	log.Infof("new subdir: %s", subexisted)
-//	subexisted, err = t.GetNvmeofSubsystem(tgtNqn)
-//	log.Infof("new subdir: %s ", subexisted)
-//	if  subexisted == "" {
-//	log.Infof("still no subsystem after add new subsystem")
-//		//t.RemoveNvmeofSubsystem(volId, tgtNqn)
-//		return errors.New("still can not get subsystem after add new one")
-//	}
-//
+	//	subexisted, err = t.GetNvmeofSubsystem(tgtNqn)
+	//	log.Infof("new subdir: %s ", subexisted)
+	//	if  subexisted == "" {
+	//	log.Infof("still no subsystem after add new subsystem")
+	//		//t.RemoveNvmeofSubsystem(volId, tgtNqn)
+	//		return errors.New("still can not get subsystem after add new one")
+	//	}
+	//
 	//create port
 	portid := t.convertTranstype(transtype)
 	portspath := NvmetDir + "/ports/" + portid

--- a/pkg/api/controllers/fileshare.go
+++ b/pkg/api/controllers/fileshare.go
@@ -28,8 +28,8 @@ import (
 	"github.com/opensds/opensds/pkg/db"
 	"github.com/opensds/opensds/pkg/model"
 	pb "github.com/opensds/opensds/pkg/model/proto"
-	. "github.com/opensds/opensds/pkg/utils/config"
 	"github.com/opensds/opensds/pkg/utils"
+	. "github.com/opensds/opensds/pkg/utils/config"
 )
 
 func NewFileSharePortal() *FileSharePortal {
@@ -384,7 +384,7 @@ func (f *FileSharePortal) DeleteFileShareAcl() {
 		log.Error("when connecting controller client:", err)
 		return
 	}
-	defer f.CtrClient.Close()	
+	defer f.CtrClient.Close()
 	metadata := utils.MergeStringMaps(fileshare.Metadata, acl.Metadata)
 
 	opt := &pb.DeleteFileShareAclOpts{

--- a/pkg/controller/controller.go
+++ b/pkg/controller/controller.go
@@ -1029,15 +1029,14 @@ func (c *Controller) DeleteFileShareAcl(contx context.Context, opt *pb.DeleteFil
 	opt.DriverName = dockInfo.DriverName
 	opt.Name = fileshare.Name
 
-	result, err := c.fileshareController.DeleteFileShareAcl((*pb.DeleteFileShareAclOpts)(opt))
-	if err != nil {
+	if err = c.fileshareController.DeleteFileShareAcl((*pb.DeleteFileShareAclOpts)(opt)); err != nil {
 		// Change the status of the file share acl to error when the creation faild
 		defer db.UpdateFileShareStatus(ctx, db.C, opt.Id, model.FileShareError)
 		log.Error("when create file share:", err.Error())
 		return pb.GenericResponseError(err), err
 	}
 
-	return pb.GenericResponseResult(result), nil
+	return pb.GenericResponseResult(nil), nil
 }
 
 // CreateFileShareSnapshot implements pb.ControllerServer.CreateFileShareSnapshot

--- a/pkg/controller/fileshare/filesharecontroller.go
+++ b/pkg/controller/fileshare/filesharecontroller.go
@@ -39,7 +39,7 @@ type Controller interface {
 
 	CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error)
 
-	DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.FileShareAclSpec, error)
+	DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error
 
 	DeleteFileShare(opt *pb.DeleteFileShareOpts) error
 
@@ -89,32 +89,25 @@ func (c *controller) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.
 
 }
 
-func (c *controller) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.FileShareAclSpec, error) {
+func (c *controller) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error {
 	if err := c.Client.Connect(c.DockInfo.Endpoint); err != nil {
 		log.Error("when connecting dock client:", err)
-		return nil, err
+		return err
 	}
 
 	response, err := c.Client.DeleteFileShareAcl(context.Background(), opt)
 	if err != nil {
 		log.Error("delete file share acl failed in file share controller:", err)
-		return nil, err
+		return err
 	}
 	defer c.Client.Close()
 
 	if errorMsg := response.GetError(); errorMsg != nil {
-		return nil,
-			fmt.Errorf("failed to delete file share acl in file share controller, code: %v, message: %v",
-				errorMsg.GetCode(), errorMsg.GetDescription())
+		return fmt.Errorf("failed to delete file share acl in file share controller, code: %v, message: %v",
+			errorMsg.GetCode(), errorMsg.GetDescription())
 	}
 
-	var fileshare = &model.FileShareAclSpec{}
-	if err = json.Unmarshal([]byte(response.GetResult().GetMessage()), fileshare); err != nil {
-		log.Error("delete file share acl failed in file share controller:", err)
-		return nil, err
-	}
-
-	return fileshare, nil
+	return nil
 }
 
 func (c *controller) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareSpec, error) {

--- a/pkg/db/drivers/etcd/etcd.go
+++ b/pkg/db/drivers/etcd/etcd.go
@@ -554,7 +554,7 @@ func (c *Client) UpdateFileShare(ctx *c.Context, fshare *model.FileShareSpec) (*
 
 	// Set update time
 	result.UpdatedAt = time.Now().Format(constants.TimeFormat)
-	
+
 	log.V(5).Infof("update file share object %+v into db", result)
 
 	body, err := json.Marshal(result)

--- a/pkg/dock/dock.go
+++ b/pkg/dock/dock.go
@@ -478,7 +478,6 @@ func (ds *dockServer) CollectMetrics(ctx context.Context, opt *pb.CollectMetrics
 	log.Infof("dock server receive CollectMetrics request, vr =%s", opt)
 
 	result, err := ds.MetricDriver.CollectMetrics()
-
 	if err != nil {
 		log.Errorf("error occurred in dock module for collect metrics: %s", err.Error())
 		return pb.GenericResponseError(err), err
@@ -512,13 +511,12 @@ func (ds *dockServer) DeleteFileShareAcl(ctx context.Context, opt *pb.DeleteFile
 
 	log.Info("dock server receive delete file share acl request, vr =", opt)
 
-	fileshare, err := ds.FileShareDriver.DeleteFileShareAcl(opt)
-	if err != nil {
+	if err := ds.FileShareDriver.DeleteFileShareAcl(opt); err != nil {
 		log.Error("when create file share in dock module:", err)
 		return pb.GenericResponseError(err), err
 	}
 	// TODO: maybe need to update status in DB.
-	return pb.GenericResponseResult(fileshare), nil
+	return pb.GenericResponseResult(nil), nil
 }
 
 // CreateFileShare implements pb.DockServer.CreateFileShare
@@ -549,7 +547,7 @@ func (ds *dockServer) DeleteFileShare(ctx context.Context, opt *pb.DeleteFileSha
 
 	log.Info("Dock server receive delete file share request, vr =", opt)
 
-	if _, err := ds.FileShareDriver.DeleteFileShare(opt); err != nil {
+	if err := ds.FileShareDriver.DeleteFileShare(opt); err != nil {
 		log.Error("error occurred in dock module when delete file share:", err)
 		return pb.GenericResponseError(err), err
 	}
@@ -582,7 +580,7 @@ func (ds *dockServer) DeleteFileShareSnapshot(ctx context.Context, opt *pb.Delet
 
 	log.Info("Dock server receive delete file share snapshot request, vr =", opt)
 
-	if _, err := ds.FileShareDriver.DeleteFileShareSnapshot(opt); err != nil {
+	if err := ds.FileShareDriver.DeleteFileShareSnapshot(opt); err != nil {
 		log.Error("error occurred in dock module when delete snapshot:", err)
 		return pb.GenericResponseError(err), err
 	}

--- a/pkg/model/proto/model.pb.go
+++ b/pkg/model/proto/model.pb.go
@@ -6,11 +6,12 @@ package proto
 import (
 	context "context"
 	fmt "fmt"
+	math "math"
+
 	proto "github.com/golang/protobuf/proto"
 	grpc "google.golang.org/grpc"
 	codes "google.golang.org/grpc/codes"
 	status "google.golang.org/grpc/status"
-	math "math"
 )
 
 // Reference imports to suppress errors if they are not otherwise used.

--- a/testutils/driver/sample.go
+++ b/testutils/driver/sample.go
@@ -122,20 +122,20 @@ func (d *Driver) DeleteVolumeGroup(opt *pb.DeleteVolumeGroupOpts) error {
 	return &model.NotImplementError{"method DeleteVolumeGroup has not been implemented yet"}
 }
 
-func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) (*model.FileShareSpec, error) {
-	return nil, nil
-}
-
 func (d *Driver) CreateFileShare(opt *pb.CreateFileShareOpts) (*model.FileShareSpec, error) {
 	return &SampleFileShares[0], nil
+}
+
+func (d *Driver) DeleteFileShare(opt *pb.DeleteFileShareOpts) error {
+	return nil
 }
 
 func (d *Driver) CreateFileShareAcl(opt *pb.CreateFileShareAclOpts) (*model.FileShareAclSpec, error) {
 	return &SampleFileSharesAcl[0], nil
 }
 
-func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) (*model.FileShareAclSpec, error) {
-	return &SampleFileSharesAcl[0], nil
+func (d *Driver) DeleteFileShareAcl(opt *pb.DeleteFileShareAclOpts) error {
+	return nil
 }
 
 // CreateFileShareSnapshot
@@ -144,6 +144,6 @@ func (d *Driver) CreateFileShareSnapshot(opt *pb.CreateFileShareSnapshotOpts) (*
 }
 
 // DeleteFileShareSnapshot
-func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) (*model.FileShareSnapshotSpec, error) {
-	return nil, nil
+func (d *Driver) DeleteFileShareSnapshot(opt *pb.DeleteFileShareSnapshotOpts) error {
+	return nil
 }

